test/functional.js

```diff
@@ -583,6 +583,50 @@ module.exports = {
             });
         });
 
+        it('When configured, TypeScript is compiled!', (done) => {
+            const config = createWebpackConfig('www/build', 'dev');
+            config.setPublicPath('/build');
+            config.addEntry('main', ['./js/render.ts', './js/index.ts']);
+            const testCallback = () => {};
+            config.configureTypeScript(testCallback);
+
+            testSetup.runWebpack(config, (webpackAssert) => {
+                // check that ts-loader transformed the ts file
+                webpackAssert.assertOutputFileContains(
+                    'main.js',
+                    'document.getElementById(\'wrapper\').innerHTML = "<h1> Hello World!</h1>";'
+                );
+
+                done();
+            });
+        });
+
+        it('When enabled, react JSX and TypeScript work nice together!', (done) => {
+            const config = createWebpackConfig('www/build', 'dev');
+            config.setPublicPath('/build');
+            config.addEntry('main', ['./js/CoolReactComponent.jsx', './js/render2.tsx']);
+            config.enableReactPreset();
+            config.configureTypeScript(function(tsConfig) {
+                tsConfig.compilerOptions = { 'jsx': 'preserve' };
```

Comment:
Can you tell me about this option? And why would react and typescript cause problems when enabled together? Is there some known issue that we want to make sure isn't a problem?